### PR TITLE
Add Backgammony

### PR DESCRIPTION
**Pull request by jonathanballs**

### CODE REVIEW

**Comment by jonathanballs** [comment]
Hello,

This is the manifest for my game Backgammony which is a Backgammon game (as you might guess).

Jonathan

**Comment by jonathanballs** [comment]
bot, build uk.jnthn.backgammony

File: uk.jnthn.backgammony.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                    "sha256": "ce1b0b0c1900717cc598032a14cf8c0ee60faf24d84368b39922c0102983bc87"
+                }
+            ]
+        },
+        {
+            "name": "backgammony",
+            "buildsystem": "simple",
+            "build-options": {
+                "append-path": "/usr/lib/sdk/dmd/bin",
+                "build-args": ["--share=network"]
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Network access during build is not allowed to ensure builds are reproducible. This is also why your build failed. There's [a dub generator](https://github.com/flatpak/flatpak-builder-tools/tree/master/dub) which can generate snippets for you.

File: uk.jnthn.backgammony.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                "./configure",
+                "make datadir=/app/share",
+                "install -D gnubg /app/bin/gnubg",
+                "install -D gnubg.wd /app/share/gnubg/gnubg.wd",
+                "install -D gnubg.weights /app/share/gnubg/gnubg.weights"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "ftp://ftp.gnu.org/gnu/gnubg/gnubg-release-1.06.002-sources.tar.gz",
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Can you switch it to https?

File: uk.jnthn.backgammony.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "org.freedesktop.Sdk.Extension.dmd"
+    ],
+    "command": "backgammony",
+    "modules": [
+        {
+            "name": "gnubg",
+            "buildsystem": "simple",
+            "build-commands": [
+                "./configure",
+                "make datadir=/app/share",
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Pass `-j$FLATPAK_BUILDER_N_JOBS` to make the build parallel, if possible.

File: uk.jnthn.backgammony.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
+    "app-id": "uk.jnthn.backgammony",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.dmd"
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Can you check if `org.freedesktop.Sdk.Extension.dmd//19.08` still works? I think it's happy coincidence it works now as 19.08 is the only branch for this extension.

File: uk.jnthn.backgammony.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            "name": "backgammony",
+            "buildsystem": "simple",
+            "build-options": {
+                "append-path": "/usr/lib/sdk/dmd/bin",
+                "build-args": ["--share=network"]
+            },
+            "build-commands": [
+                "dub build",
+                "install -D backgammony /app/bin/backgammony",
+                "install -d \"/app/share/applications\"",
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: I don't think there's a need to additionally quote strings which won't have any variables interpolated by shell.

File: uk.jnthn.backgammony.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                "install -D resources/linux/uk.jnthn.backgammony.desktop /app/share/applications",
+                "install -d \"/app/share/icons/hicolor/512x512/apps\"",
+                "install -D resources/icon.png /app/share/icons/hicolor/512x512/apps/uk.jnthn.backgammony.png",
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Any chance for SVG icon as well? (Just curious!)

File: uk.jnthn.backgammony.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                "install -d /app/share/metainfo",
+                "install -D resources/linux/uk.jnthn.backgammony.appdata.xml /app/share/metainfo/uk.jnthn.backgammony.metainfo.xml"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/jonathanballs/backgammon.git",
+                    "branch": "master"
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Please change it to specific tag or commit.

File: uk.jnthn.backgammony.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                }
+            ]
+        }
+    ],
+    "finish-args": [
```
Comment (barthalion) [inline uk.jnthn.backgammony.json]: Can you move it above `modules` array? It's a nitpick, but improves readability a lot.

**Comment by jonathanballs** [comment]
Thank you, I will get these fixed 😅

File: uk.jnthn.backgammony.json (hunk 3 above)
Comment (gasinvein) [reply uk.jnthn.backgammony.json]: Why even use `buildsystem: simple` here? It looks like generic autotools.

File: uk.jnthn.backgammony.json (hunk 4 above)
Comment (jonathanballs) [reply uk.jnthn.backgammony.json]: @barthalion sorry would you be able to expand on what you mean by this?

**Comment by jonathanballs** [comment]
bot, build uk.jnthn.backgammony

**Comment by gasinvein** [comment]
@jonathanballs You don't need to copy-paste generated sources to your manifest, just drop the `generated-sources.json` file into this repo and add it's filename after your app's sources, like so:
```json
"sources": [
  {
    "type": "git",
    ...
  },
  "generated-sources.json"
]
```

File: uk.jnthn.backgammony.json (hunk 4 above)
Comment (gasinvein) [reply uk.jnthn.backgammony.json]: @barthalion The extension point in Gnome SDK 3.34 explicitly sets it's version to `19.08`:
```ini
[Runtime]
name = org.gnome.Sdk
runtime = org.gnome.Platform/x86_64/3.34
sdk = org.gnome.Sdk/x86_64/3.34

...

[Extension org.freedesktop.Sdk.Extension]
subdirectories = true
directory = lib/sdk
no-autodownload = true
add-ld-path = lib
version = 19.08
```

**Comment by jonathanballs** [comment]
bot, build uk.jnthn.backgammony

**Comment by jonathanballs** [comment]
bot, build uk.jnthn.backgammony

**Comment by gasinvein** [comment]
@jonathanballs `dmd` is unavailable for ARM, you can limit build arches to x86-64 by creating a file `flathub.json` with contents `{ "only-arches": ["x86_64"] }`.

**Comment by jonathanballs** [comment]
bot, build uk.jnthn.backgammony

**Comment by jonathanballs** [comment]
@gasinvein I've tested using the test repo and I'm happy to submit this.

**Comment by gasinvein** [comment]
@jonathanballs I'm not a Flathub admin - you should now wait for one of them to re-review.

File: uk.jnthn.backgammony.json (hunk 4 above)
Comment (barthalion) [reply uk.jnthn.backgammony.json]: @gasinvein I don't think it matters at all but let's worry in 6 months.

**Pull request closed without merging**